frontend/apps/filemanager/filemanagersetdefaults.lua

```diff
@@ -102,264 +79,243 @@ function SetDefaults:init()
         end
     end
     self.defaults_menu:swithItemTable("Defaults", self.results)
     UIManager:show(menu_container)
 end
 
 function SetDefaults:close()
-    self.set_dialog:onClose()
     UIManager:close(self.set_dialog)
 end
 
 function SetDefaults:ConfirmSave()
     UIManager:show(ConfirmBox:new{
-        text = _("Are you sure you want to save the settings to \"defaults.persistent.lua\"?"),
+        text = _('Are you sure you want to save the settings to "defaults.persistent.lua"?'),
         ok_callback = function()
             self:SaveSettings()
         end,
     })
 end
 
 function SetDefaults:build_setting(j)
-    local ret = self.defaults_name[j] .. " = "
-    if type(_G[self.defaults_name[j]]) == "boolean" or type(_G[self.defaults_name[j]]) == "table" then
-        ret = ret .. tostring(self.defaults_value[j])
+    local setting_name = self.defaults_name[j]
+    local ret = setting_name .. " = "
+    if type(_G[setting_name]) == "boolean" then
+        return ret .. tostring(self.defaults_value[j])
+    elseif type(_G[setting_name]) == "table" then
+        return ret .. "{...}"
     elseif tonumber(self.defaults_value[j]) then
-        ret = ret .. tostring(tonumber(self.defaults_value[j]))
+        return ret .. tostring(tonumber(self.defaults_value[j]))
     else
-        ret = ret .. "\"" .. tostring(self.defaults_value[j]) .. "\""
+        return ret .. "\"" .. tostring(self.defaults_value[j]) .. "\""
     end
-    return ret
 end
 
 function SetDefaults:SaveSettings()
-
-    local function fileread(filename,array)
-        local file = io.open(filename)
-        local line = file:read()
-        local counter = 0
-        while line do
-            counter = counter + 1
-            local i = string.find(line,"[-][-]") -- remove comments from file
-            if (i or 0)>1 then line = string.sub(line,1,i-1) end
-            array[counter] = line:gsub("^%s*(.-)%s*$", "%1") -- trim
-            line = file:read()
-        end
-        file:close()
-    end
-
+    self.results = {}
     local persistent_filename = DataStorage:getDataDir() .. "/defaults.persistent.lua"
     local file
-    if io.open(persistent_filename,"r") == nil then
-        file = io.open(persistent_filename, "w")
-        file:write("-- For configuration changes that persists between (nightly) releases\n")
+    local persisted_defaults = {}
+    file = io.open(persistent_filename,"r")
+    if file ~= nil then
         file:close()
+        local load_defaults = loadfile(persistent_filename)
+        setfenv(load_defaults, persisted_defaults)
+        load_defaults()
     end
 
-    local dpl = {}
-    fileread(persistent_filename, dpl)
-    local dl = {}
-    fileread("defaults.lua",dl)
-    self.results = {}
-    local done = {}
-
-    for j=1,#SetDefaults.defaults_name do
-        if not self.changed[j] then done[j] = true end
+    local checked = {}
+    for j=1, #self.defaults_name do
+        if not self.changed[j] then checked[j] = true end
```

Comment:
Should we check whether the value is consistent with 'default' default value?
i.e. User may set some value back to 'default' default value, we should eventually remove them.